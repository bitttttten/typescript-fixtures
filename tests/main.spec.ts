import { test, expect } from "vitest";
import { parse } from "graphql";
import { plugin } from "../src/index.js";

test("Should generate mocks based on queries and mutations", async () => {
  const documents = [
    {
      document: parse(
        `query User { slug id firstName lastName user { address { number street }} }`
      ),
    },
  ];

  const result = await plugin(null, documents, {});

  // handler function names
  expect(result.content).toContain(`generateUserQueryFixture`);
  expect(result.content).toMatchSnapshot("content");

  // handles imports
  expect(result.prepend).toMatchSnapshot("imports");
});
