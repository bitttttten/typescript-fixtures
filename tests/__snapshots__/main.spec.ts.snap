// Vitest Snapshot v1

exports[`Should generate mocks based on queries and mutations > content 1`] = `
"

export const generateUserQueryFixture = (user?: Partial<UserQuery>) {
	const slug = faker.lorem.slug();
	const id = faker.datatype.uuid();
	const firstName = faker.name.firstName();
	const lastName = faker.name.lastName();
	const user = {
		address: {
			number: faker.datatype.number(),
			street: faker.address.street(),
		}
	}

	const base = {
		slug,
		id,
		firstName,
		lastName,
		user,
	}

	return merge(base, user)
}
          "
`;

exports[`Should generate mocks based on queries and mutations > imports 1`] = `
[
  "import { faker } from '@faker-js/faker'",
]
`;
